finality-verifiers/grandpa/src/side_effects.rs

```diff
@@ -0,0 +1,281 @@
+use crate::{Config, Error};
+use codec::{Decode, Encode};
+use sp_runtime::DispatchError;
+use sp_std::{vec, vec::Vec};
+
+#[derive(Encode, Decode)]
+pub enum TransferEventStub<T: frame_system::Config, Balance> {
+    Endowed(T::AccountId, Balance),
+    DustLost(T::AccountId, Balance),
+    Transfer {
+        from: T::AccountId,
+        to: T::AccountId,
+        amount: Balance,
+    },
+}
+
+type CurrencyId = u32;
+
+#[derive(Encode, Decode)]
+pub enum MultiTransferEventStub<T: frame_system::Config, Balance, CurrencyId> {
+    Endowed(CurrencyId, T::AccountId, Balance),
+    DustLost(CurrencyId, T::AccountId, Balance),
+    Transfer {
+        currency_id: CurrencyId,
+        from: T::AccountId,
+        to: T::AccountId,
+        amount: Balance,
+    },
+}
+
+pub(crate) fn decode_event<T: Config<I>, I: 'static>(
```

Comment:
So eth events decoder removed?

Reply:
Oui, will add to the eth2 pallet once ready